src/bartiq/compilation/_compile.py

```diff
@@ -101,22 +113,40 @@ def _compile_constraint(
 
 
 def compile_routine(
-    routine: SchemaV1 | RoutineV1,
+    routine: SchemaV1 | RoutineV1 | Routine[T],
     *,
     backend: SymbolicBackend[T] = sympy_backend,
-    precompilation_stages: Iterable[PreprocessingStage[T]] = DEFAULT_PRECOMPILATION_STAGES,
+    preprocessing_stages: Iterable[PreprocessingStage[T]] = DEFAULT_PREPROCESSING_STAGES,
     skip_verification: bool = False,
 ) -> CompilationResult[T]:
-    if not skip_verification:
+    """Performs symbolic compilation of a given routine.
+
+    Compilation in this context means taking a routine that's defined in terms of variables defined within the scope
+    of a given routine and returning a routinedefined in terms of top-level variables.
```

Comment:
I would slightly alter this description to be more concise (and arguably - more understandable). WDYT about the following:

```suggestion
    In this context, compilation means transforming a routine defined in terms of routine-local variables into one defined in terms of global input parameters.
```